src/Sentry/SentryPropagationContext.cs

```diff
@@ -0,0 +1,58 @@
+using Sentry.Extensibility;
+
+namespace Sentry;
+
+internal class SentryPropagationContext
+{
+    public SentryId TraceId { get; }
+    public SpanId SpanId { get; }
+    public SpanId? ParentSpanId { get; }
+    public DynamicSamplingContext? DynamicSamplingContext { get; set; }
+
+    internal static IDiagnosticLogger? Logger { get; set; } = SentrySdk.CurrentOptions?.DiagnosticLogger;
```

Comment:
Can you pass explicitly the logger instead of statically accessing it? The whole design avoided static access to make things easier to reason about and unit test.

The class is internal so we could just take the logger as an argument on methods that need it, have the caller provide it?

I might be too pedantic here since this is all internal code we can change anytime so if it works, it works, but I'd at least suggest:

```suggestion
    internal static IDiagnosticLogger? Logger => SentrySdk.CurrentOptions?.DiagnosticLogger;
```

So it picks up on changes to the logger implementation at runtime.